src/test/scala/PropertyCheckerTest.scala

```diff
@@ -0,0 +1,464 @@
+package CSVValidation
+import com.fasterxml.jackson.databind.node.{ArrayNode, BooleanNode, IntNode, JsonNodeFactory, NullNode, ObjectNode, TextNode, ValueNode}
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import org.scalatest.{BeforeAndAfter, FunSuite, Tag}
+
+
+class PropertyCheckerTest extends FunSuite {
+  val objectMapper = new ObjectMapper()
+
+  test("boolean property checker should return correct types") {
+    val result = PropertyChecker.checkProperty("suppressOutput", new TextNode("w3c"), "https://www.w3.org/", "und")
+    assert(result._1.isInstanceOf[BooleanNode])
+    assert(result._2.isInstanceOf[Array[String]])
+    assert(result._3.isInstanceOf[PropertyType.Value])
+  }
+
+  test("boolean property checker should return invalid for non boolean values") {
+    val (value, warnings, typeString) = PropertyChecker.checkProperty("suppressOutput", new TextNode("w3c"), "https://www.w3.org/", "und")
+    assert(warnings === Array[String]("invalid_value"))
+    assert(value === BooleanNode.getFalse)
+  }
+
+  test("boolean property checker should return zero warnings for boolean values") {
+    val (value, warnings, typeString) = PropertyChecker.checkProperty("suppressOutput", BooleanNode.getTrue, "https://www.w3.org/", "und")
+    assert(value === BooleanNode.getTrue)
+    assert(warnings.isEmpty)
+  }
+
+  test("language property checker should return invalid for properties which cannot be accepted") {
+    val (_, warnings, _) = PropertyChecker.checkProperty("@language", new TextNode("Invalid Language Property"), "https://www.w3.org/", "und")
+    assert(warnings === Array[String]("invalid_value"))
+  }
+
+  test("language property checker should return no warnings for correct language property") {
+    val (_, warnings, _) = PropertyChecker.checkProperty("@language", new TextNode("sgn-BE-FR"), "https://www.w3.org/", "und")
```

Comment:
I'm still not sure what this "und" value means. Any ideas?

Reply:
I dont think und has any special significance. The ruby code has checking for this lang string at some places so I decided to continue using this value for lang